gtdynamics/kinematics/KinematicsSlice.cpp

```diff
@@ -51,6 +51,22 @@ NonlinearFactorGraph Kinematics::graph<Slice>(const Slice& slice,
   return graph;
 }
 
+template <>
+EqualityConstraints Kinematics::constraints<Slice>(const Slice& slice,
+                                              const Robot& robot) const {
+  EqualityConstraints constraints;
+
+  // Constrain kinematics at joints.
+  for (auto&& joint : robot.joints()) {
+    auto constraint_expr = joint->poseConstraint(slice.k);
+    gtsam::Vector6 tolerance = p_->p_cost_model->sigmas();
```

Comment:
cost models should now be replaced with tolerances I think. But we can do that in a different PR

Reply:
added a TODO